contrib/py/admin/dumpstats.py

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env python3
+import requests
+import json
+import time
+import curses
+import math
+import traceback
+
+
+class Monitor:
+
+    _speedSamples = 8
+    _globalspeed = []
+
+    def __init__(self, url):
+        self.data = dict()
+        self.win = curses.initscr()
+        self._url = url
+        while len(self._globalspeed) < self._speedSamples:
+            self._globalspeed.append((0, 0))
+
+    def __del__(self):
+        curses.endwin()
+
+    def on_timer(self, event):
+        """called on timer event"""
+        self.update_data()
+
+    def jsonrpc(self, meth, params):
+        r = requests.post(
+            self._url,
+            headers={"Content-Type": "application/json"},
+            json={
+                "jsonrpc": "2.0",
+                "id": "0",
+                "method": "{}".format(meth),
+                "params": params,
+            },
+        )
+        return r.json()
+
+    def update_data(self):
+        """update data from lokinet"""
+        try:
+            j = self.jsonrpc("llarp.admin.dumpstate", {})
+            self.data = j["result"]
+        except Exception as ex:
+            self.data = None
+
+    def _render_path(self, y, path, name):
+        """render a path at current position"""
+        self.win.move(y, 1)
+        self.win.addstr("({}) ".format(name))
+        y += 1
+        self.win.move(y, 1)
+        y += 1
+        self.win.addstr("me -> ")
+        for hop in path["hops"]:
+            self.win.addstr(" {} ->".format(hop["router"][:4]))
+        self.win.addstr(" [{} ms latency]".format(path["intro"]["latency"]))
+        self.win.addstr(" [{} until expire]".format(self.timeTo(path["expiresAt"])))
+        if path["expiresSoon"]:
+            self.win.addstr("(expiring)")
+        elif path["expired"]:
+            self.win.addstr("(expired)")
+        return y
+
+    def timeTo(self, ts):
+        """ return time until timestamp in seconds formatted"""
+        now = time.time() * 1000
+        return "{} seconds".format(int((ts - now) / 1000))
+
+    def speedOf(self, rate):
+        """turn int speed into string formatted"""
+        units = ["B", "KB", "MB", "GB"]
+        idx = 0
+        while rate > 1000 and idx < len(units):
+            rate /= 1000.0
+            idx += 1
+        return "{} {}ps".format("%.2f" % rate, units[idx])
+
+    def display_service(self, y, name, status):
+        """display a service at current position"""
+        self.win.move(y, 1)
+        self.win.addstr("service [{}]".format(name))
+        build = status["buildStats"]
+        ratio = build["success"] / (build["attempts"] or 1)
```

Comment:
I'm not sure what will be in `build` here, but if `"attempts"` is not present this will throw a KeyError.  It's fine if it'll be present and falsey (False, None, 0), but if it can be not present you'd need something (ugly) like:
```Python
... / (build["attempts"] or 1 if "attempts" in build else 1)
```
or else set it beforehand.  (But maybe it's always set so this doesn't matter).

Reply:
the `or 1` is to prevent division by 0